interfaces/Glitter/templates/static/javascripts/glitter.history.js

```diff
@@ -257,6 +259,89 @@ function HistoryListModel(parent) {
             $("#modal-purge-history").modal('hide');
         });
     };
+
+    // Show the input checkbox
+    self.showMultiEdit = function() {
+        self.isMultiEditing(!self.isMultiEditing())
+        self.multiEditItems.removeAll();
+        $('.history-table input[name="multiedit"], #history-options #multiedit-checkall').prop({'checked': false, 'indeterminate': false})
+    }
+
+    // Add to the list
+    self.addMultiEdit = function(item, event) {
+        if(event.shiftKey) {
+            checkShiftRange('.history-table input[name="multiedit"]');
+        }
+
+        if(event.currentTarget.checked) {
+            self.multiEditItems.push(item);
+        } else {
+            self.multiEditItems.remove(function(inList) { return inList.nzo_id == item.nzo_id; })
+        }
+
+        setCheckAllState('#history-options #multiedit-checkall', '.history-table input[name="multiedit"]')
+        return true;
+    }
+
+
+    // Check all
+    self.checkAllJobs = function(item, event) {
```

Comment:
Is there a way we can generalize this function so we can also use it for the queue? There is a lot of code duplication here, some of which is needed, but I think this function could be generalized.

Reply:
Sure, I'll take a look at it again once I have time and will try to reduce duplication as much as possible.